packages/webapp/src/containers/Finances/Report/index.jsx

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { ReactComponent as ReportIcon } from '../../../assets/images/finance/Report-icn.svg';
+import FinanceDateRangeSelector from '../../../components/Finances/DateRangeSelector';
+import Button from '../../../components/Form/Button';
+import TextButton from '../../../components/Form/Button/TextButton';
+import ModalComponent from '../../../components/Modals/ModalComponent/v2';
+import { Semibold, Text } from '../../../components/Typography';
+import TransactionFilterContent from '../../Filter/Transactions';
+import { EXPENSE_TYPE, REVENUE_TYPE } from '../../Filter/constants';
+import { transactionsFilterSelector } from '../../filterSlice';
+import { downloadFinanceReport } from '../saga';
+import { dateRangeDataSelector } from '../selectors';
+import useTransactions from '../useTransactions';
+import styles from './styles.module.scss';
+
+const Report = () => {
+  const { t } = useTranslation();
+
+  const dateRange = useSelector(dateRangeDataSelector);
+  const defaultTransactionsFilter = useSelector(transactionsFilterSelector);
+
+  const [isExportReportOpen, setIsExportReportOpen] = useState(false);
+  const [dateFilter, setDateFilter] = useState(dateRange);
+
+  const dispatch = useDispatch();
+  const filterRef = useRef({});
+  const transactions = useTransactions({
+    dateFilter,
+    expenseTypeFilter: filterRef.current[EXPENSE_TYPE],
```

Comment:
Reading first is the way! 😅 In testing this I do see the filters being applied correctly -- I think if the ref value changes, that new value will still get passed to the transactions hook and trigger the memo. Can you let me know though if you are seeing any problems with the filters?

That being said, we **should** be using state variables instead of refs in filters altogether. I've also seen the same with dropdowns, where we're passing a ref instead of a state variable. In general we shouldn't be using refs unless absolutely necessary for something that needs to happen outside of the normal React flow, but since it's a broad issue we'll have to tackle it as a refactor.

Reply:
Interesting! No I definitely double-checked the behaviour (I don't trust my reading skills _that_ much yet 😂) and saw the update issue in testing as well. Is this different than what you see? I am logging in the saga:

https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/e0ef52be-7899-4fe5-ac39-87a663ea4fba

For me, the data sent 1) doesn't update based on edits made in the modal after loading it, 2) but are applied after the modal is reloaded (minor bug is that the visual filter state appears to be reset to what the dashboard shows but the data is using the filter setting from the last modal edit!)